tests/integration/clients/test_websocket_client.py

```diff
@@ -0,0 +1,52 @@
+"""Performance testing of Websocket clients."""
+
+from __future__ import annotations
+
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import test_async_and_sync
+from xrpl.models.currencies import XRP, IssuedCurrency
+from xrpl.models.requests import BookOffers
+from xrpl.models.response import ResponseStatus
+
+
+class TestWebsocketClient(IntegrationTestCase):
+    """Memory usage of websocket client"""
+
+    @test_async_and_sync(globals(), websockets_only=True)
```

Comment:
This test is against standalone, not mainnet. The data isn't going to really test anything.

Reply:
Testing with a Standalone mode provides a useful baseline, because the performance will only get worse with Mainnet/Testnet.

I'm measuring the memory usage of the client when multiple successive requests are input to the client. (Since no new transactions are provided, there is no need to advance the ledger)

I can set `use_testnet=True`, I'm not opposed to that.

Reply:
The current integration test framework does not easily let us test against the mainnet (unless we write additional utility methods). I'm assuming you're referring to the `use_testnet` parameter in `test_async_and_sync` decorator.

Reply:
The integration test framework can be changed, it's not set in stone.

Reply:
are you suggesting that I test the performance against the mainnet? Why would that be different than testing with the testnet or the standalone node?

The performance of the xrpl-py client is in question, does it matter if I use a standalone/testnet/mainnet server for this test?

Reply:
There isn't enough traffic on testnet/standalone to test the performance of xrpl-py, it's not going to run into any issues. There may not even be any data in this pair (or any pair) for the few seconds that you're testing on. You could theoretically generate that traffic yourself, but that'd require a much more complex test.